python/lsst/ip/diffim/utils.py

```diff
@@ -711,3 +717,361 @@ def plotWhisker(results, newWcs):
+        negDetectedBits = negImage.getMaskedImage().getMask().getArray() == dm.getPlaneBitMask("DETECTED")
+        pos_det = dm.addMaskPlane("DETECTED_POS")  # new mask plane -- different from "DETECTED"
+        neg_det = dm.addMaskPlane("DETECTED_NEG")  # new mask plane -- different from "DETECTED_NEGATIVE"
+        dma = dm.getArray()
+        # set the two custom mask planes to these new masks
+        dma[:, :] = posDetectedBits*pos_det + negDetectedBits*neg_det
+        return dipole, (posImage, posCatalog), (negImage, negCatalog)
+
+    def fitDipoleSource(self, source, **kwds):
+        alg = dipoleFitTask.DipoleFitAlgorithm(self.diffim, self.posImage, self.negImage)
+        fitResult = alg.fitDipole(source, **kwds)
+        return fitResult
+
+    def displayImages(self):
+        displayExposure(self.diffim)
+        displayExposure(self.posImage)
+        displayExposure(self.negImage)
+
+    def displayCutouts(self, source, asHeavyFootprint=True):
+        displayCutouts(source, self.diffim, self.posImage, self.negImage, asHeavyFootprint)
+
+    def detectDipoleSources(self, doMerge=True, diffim=None, detectSigma=5.5, grow=3):
+        """Utility function for detecting dipoles.
+
+        Detect pos/neg sources in the diffim, then merge them. A
+        bigger "grow" parameter leads to a larger footprint which
+        helps with dipole measurement for faint dipoles.
+
+        """
+
+        if diffim is None:
+            diffim = self.diffim
+
+        # Start with a minimal schema - only the fields all SourceCatalogs need
+        schema = afwTable.SourceTable.makeMinimalSchema()
+
+        # Customize the detection task a bit (optional)
+        detectConfig = measAlg.SourceDetectionConfig()
+        detectConfig.returnOriginalFootprints = False  # should be the default
+
+        psfSigma = diffim.getPsf().computeShape().getDeterminantRadius()
+
+        # code from imageDifference.py:
+        detectConfig.thresholdPolarity = "both"
+        detectConfig.thresholdValue = detectSigma
+        # detectConfig.nSigmaToGrow = psfSigma
+        detectConfig.reEstimateBackground = True  # if False, will fail often for faint sources on gradients?
+        detectConfig.thresholdType = "pixel_stdev"
+
+        # Create the detection task. We pass the schema so the task can declare a few flag fields
+        detectTask = measAlg.SourceDetectionTask(schema, config=detectConfig)
+
+        table = afwTable.SourceTable.make(schema)
+        catalog = detectTask.makeSourceCatalog(table, diffim, sigma=psfSigma)
+
+        # Now do the merge.
+        if doMerge:
+            fpSet = catalog.fpSets.positive
+            fpSet.merge(catalog.fpSets.negative, grow, grow, False)
+            sources = afwTable.SourceCatalog(table)
+            fpSet.makeSources(sources)
+
+            return sources
+
+        else:
+            return detectTask, schema
+
+
+# ######## UTILITIES FUNCTIONS FOR PLOTTING  ####
```

Comment:
Ok, and there are multiple ways for importing `matplotlib` in that file. Adding a special `importMatplotlib` function halfway through is going to be confusing unless it's at the top of the file and all the plot routines use it.